Makefile.include

```diff
@@ -242,6 +242,16 @@ ifneq (, $(filter $(GLOBAL_GOALS), $(MAKECMDGOALS)))
   include $(RIOTMAKE)/buildtests.inc.mk
 else
 
+
+.PHONY: CREATECACHEDIR
+
+CREATECACHEDIR:
```

Comment:
such comfort targets are done by:

```
CREATECACHEDIR: $(BUILD_DIR)/CACHEDIR.TAG
```
just depend on what you want to have build

not sure if this case makes sense

Reply:
I concur with the above.

Reply:
I edited it to keep the `$(BUILD_DIR)/CACHEDIR.TAG` target. 

And no, I'm not replacing the target `all`. The includes means that regardless of the target called originally, the `CREATECACHEDIRS` will also be called by default.

The problem with using `all` is that you have to specify `CREATECACHEDIRS` as a dependency for every single target that will use it. (which is very uncomfortable for me tbh)

Reply:
The default is not about the include but about being the first non dot rule in a makefile which is the default if you call just `$ make ` previously it was `all`, which also include with you rule above it it is create a directory.

the include is just a strange way to enforce something happening (a hopefully allways failing try of an include)

https://www.gnu.org/software/make/manual/make.html#How-Make-Works

> By default, make starts with the first target (not targets whose names start with ‘.’ unless they also contain one or more ‘/’). This is called the default goal. 

https://www.gnu.org/software/make/manual/make.html#Standard-Targets
> All GNU programs should have the following targets in their Makefiles: 
> ‘all’ ... Compile the entire program. This should be the default target.

There probably is a reason you are doing these strange things so maybe you can enlighten me -- Why is it uncomfotable to have the directory created when it is needed?

Reply:
As I said, it could be added in the targets that needs a cache directory and work as well but it means that whenever a target would need this dependency it will need to include it in its prerequisites. I don't think that creating empty directories in advance (that will most likely be filled soon) is costing that much. 

Also the case I mentionned, `pkg-build` was using `$(BUILD_DIR)/CACHEDIR.TAG` but not `pkg-prepare` when it was the target that actually fills the directory. With this specific case in mind I thought it was better to call `CREATECACHEDIRS` once by default rather than calling it twice (once in each target).